### DISCUSSION TITLE
Script compiled with pyinstaller is missing a .dll file, when the file is manually copied in the program's folder it just dies

### DISCUSSION THREAD

**Discussion by that-ugly-cat** [pyinstaller/pyinstaller#6563] [Help]
Hello! 

I have a python script which is basically a graphic interface (pysimpleguy) to a mysql database.

I am working in windows 10, python 3.8; my dependencies are:

- PySimpleGUI 4.55.1
- sqlalchemy 1.3.20
- pymysql 1.0.2
- pandas 1.1.3
- regex 2020.10.15
- pillow 8.0.1

The code works and I'd like to compile it to .exe to distribute it to users in my organization.

I tried to compile it with:

`pyinstaller -D .\db_interface_v3.6.1_release.py --debug=imports
`
However, pyinstaller throws some errors when compiling:

```
201667 INFO: Building COLLECT COLLECT-00.toc
Traceback (most recent call last):
  File "c:\users\spit\anaconda3\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\spit\anaconda3\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\Spit\anaconda3\Scripts\pyinstaller.exe\__main__.py", line 7, in <module>
  File "c:\users\spit\anaconda3\lib\site-packages\PyInstaller\__main__.py", line 124, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "c:\users\spit\anaconda3\lib\site-packages\PyInstaller\__main__.py", line 58, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "c:\users\spit\anaconda3\lib\site-packages\PyInstaller\building\build_main.py", line 782, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "c:\users\spit\anaconda3\lib\site-packages\PyInstaller\building\build_main.py", line 714, in build
    exec(code, spec_namespace)
  File "C:\Users\Spit\Desktop\DIPEx db parser\db_interface_v3.6.1_release.spec", line 37, in <module>
    coll = COLLECT(exe,
  File "c:\users\spit\anaconda3\lib\site-packages\PyInstaller\building\api.py", line 818, in __init__
    self.__postinit__()
  File "c:\users\spit\anaconda3\lib\site-packages\PyInstaller\building\datastruct.py", line 155, in __postinit__
    self.assemble()
  File "c:\users\spit\anaconda3\lib\site-packages\PyInstaller\building\api.py", line 866, in assemble
    shutil.copy(fnm, tofnm)
  File "c:\users\spit\anaconda3\lib\shutil.py", line 415, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "c:\users\spit\anaconda3\lib\shutil.py", line 261, in copyfile
    with open(src, 'rb') as fsrc, open(dst, 'wb') as fdst:
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Spit\\Desktop\\DIPEx db parser\\dist\\db_interface_v3.6.1_release\\share\\jupyter\\lab\\staging\\node_modules\\.cache\\terser-webpack-p
lugin\\content-v2\\sha512\\2e\\ba\\cfce62ec1f408830c0335f2b46219d58ee5b068473e7328690e542d2f92f2058865c600d845a2e404e282645529eb0322aa4429a84e189eb6b58c1b97c1a'
```
If I try to run the compiled exe, I get an error regarding a specific .dll:

```
INTEL MKL ERROR: Impossibile trovare il modulo specificato. mkl_intel_thread.dll.
Intel MKL FATAL ERROR: Cannot load mkl_intel_thread.dll.
```
If I take this missing .dll from my Anaconda environment and copy it into the program's folder, when I try to run the .exe again it just dies without further messages:

```
import 'numpy.ma' # <pyimod03_importers.FrozenImporter object at 0x000001F6A455BEE0>
PS C:\Users\Spit\Desktop\DIPEx db parser\dist\db_interface_v3.6.1_release>
```
Any idea on how to sort it out?

Thanks!

**Comment by bwoodsend** [comment]
> FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\Spit\\Desktop\\DIPEx db parser\\dist\\db_interface_v3.6.1_release\\share\\jupyter\\lab\\staging\\node_modules\\.cache\\terser-webpack-p
lugin\\content-v2\\sha512\\2e\\ba\\cfce62ec1f408830c0335f2b46219d58ee5b068473e7328690e542d2f92f2058865c600d845a2e404e282645529eb0322aa4429a84e189eb6b58c1b97c1a'

I don't suppose you've forgotten to disable [Windows's PATH_MAX limitation](https://docs.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=cmd)?

> If I try to run the compiled exe, I get an error regarding a specific .dll:

PyInstaller does not really support Anaconda for precisely this reason. What it does to DLLs is very hard for PyInstaller track. If you're going to use PyInstaller then either install a normal version of Python or at least create a `venv` environment and install packages only with pip.

**Reply by rokm** [reply]
This indeed looks like max path limitation. Or possibly a python version that does not opt in to long paths (e.g., affected by [python 45777](https://bugs.python.org/issue45777), although that does not list python 3.8 as affected version).

> If I try to run the compiled exe, I get an error regarding a specific .dll:
> If I take this missing .dll from my Anaconda environment and copy it into the program's folder, when I try to run the .exe again it just dies without further messages:

You're trying to run an incompletely collected application, in which more than one DLL is missing. So you'll need to sort out the original error and get a properly collected application.

**Reply by that-ugly-cat** [reply]
Yup, makes sense. Could you please provide some more references regarding the max path issue? Thanks!

**Reply by bwoodsend** [reply]
Just search _Windows 260 character path_ limit if that msdocs link isn't enough.

**Reply by rokm** [reply]
> Could you please provide some more references regarding the max path issue? Thanks!

Is the MSDN page that Brénainn linked not enough? It contains a section on how to enable long paths via registry key.

**Reply by that-ugly-cat** [reply]
got it - thanks, I didn't notice the link, my bad. Changed that setting, trying to compile again right now. Will let you know whether it works - else, I'll set up a non-anaconda environment and try again. Thanks!

**Reply by that-ugly-cat** [reply]
worked like a charm - thanks!
